### ISSUE TITLE
Using `-Z macro-backtrace` breaks 'unreachable statement' with macro call

### ISSUE THREAD

**Issue by Aaron1011** [rust-lang/rust#73700]
The following code:

```rust
fn main() {
    panic!("{}", 25);
    return ();
}
```

prints the following warning on nightly:

```
warning: unreachable statement
 --> panic_unreachable.rs:3:5
  |
2 |     panic!("{}", 25);
  |     ----------------- any code following this expression is unreachable
3 |     return ();
  |     ^^^^^^^^^^ unreachable statement
  |
  = note: `#[warn(unreachable_code)]` on by default
```

However, with `-Z macro-backtrace`, this warning is printed:

```
warning: unreachable statement
  --> panic_unreachable.rs:3:5
   |
3  |     return ();
   |     ^^^^^^^^^^ unreachable statement
   | 
  ::: /home/aaron/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/macros.rs:16:9
   |
16 |         $crate::rt::begin_panic_fmt(&$crate::format_args!($fmt, $($arg)+))
   |         ------------------------------------------------------------------ any code following this expression is unreachable
   |
   = note: `#[warn(unreachable_code)]` on by default

```

The original `panic!` call isn't even shown, making the "any code following this expression is unreachable" almost useless.

**Comment by oscarsun95** [comment]
I'm also trying to use -Z macro-bactrace (in nightly build).
Could you show your full command please?

**Comment by Aaron1011** [comment]
@oscarsun95: I used `rustc main.rs -Z macro-backtrace`

**Comment by RodBurman** [comment]
This issue remains in rustc 1.100.0-nightly (c54751567 2026-08-22).
